editor/src/clj/editor/ui.clj

```diff
@@ -759,27 +768,24 @@
   HasAction
   (on-action! [node update-fn]
     (.setOnAction node (event-handler e
-                         ;; Clear the auto-commit flag. Further edits will re-apply it.
-                         (when (user-data node ::auto-commit)
-                           (user-data! node ::auto-commit false))
+                         (clear-auto-commit! node)
+                         (update-fn e)
                          (if (zero? (.getLength (.getSelection node)))
                            (.selectAll node)
-                           (.deselect node))
-                         (update-fn e))))
+                           (.deselect node)))))
   Cancellable
   (on-cancel! [node cancel-fn]
     (bind-key! node "Esc" (fn []
                             (cancel-fn node)
+                            (clear-auto-commit! node)
```

Comment:
I checked with debug prints, and we should clear the auto-commit flag after calling the `cancel-fn`. If we do not, we will commit the field value when the `.requestFocus` call resolves. The property editor doesn't create an additional undo step because it also compares the property values against the old values internally, but we want to be able to use `on-cancel!` elsewhere too!

Reply:
I wanted to add it but stuck with `fn` issue and forgot =/